lib/perl/OVH/Bastion/Plugin.pm

```diff
@@ -40,6 +38,27 @@ sub begin {
 
     $helpfunc = \&help if (ref $helpfunc ne 'CODE');
 
+    # our parent (osh.pl) is responsible for ensuring the closing log is written,
+    # we don't have to handle it here, however if we get a signal, ensure that we propagate it to our
+    # potential children, through the process group, so that no dangling process is left behind.
+    # by default we want to continue what we were doing (hence the return), except if exitOnSignal is specified.
+    $SIG{$_} = sub {
```

Comment:
TBH that's quite strange to read with a statement modfier

Reply:
This func was way simpler when I added the statement modifier, now I reckon that it's kinda long and the statement modifier ends up being too far away from the beginning of the block. Fixing that.